### PR TITLE
0040 remove password app

### LINKED ISSUE #39: Allow users to change the mnemonic password
Users should be allowed to change their mnemonic password without having to go through the reset process.

### PR DESCRIPTION
**Pull request by DSMatusevich**
**Description**:
- The application password was removed. 
- The recover password workflow was removed
- Added the option to change the mnemonic password
- Added a workflow to recover the mnemonic password
- Added a workflow to recover a key password (can be different)
- Added error checking for when the user enters a mnemonic different from the one stored.

**Related issue(s)**:

Closes #39, #40 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by abhishek-hedera — APPROVED** [review]

File: tools-ui/src/main/java/com/hedera/hashgraph/client/ui/utilities/KeyPairUtility.java [hunk 1]
```diff
@@ -73,18 +89,61 @@ private KeyPair getKeyPair(File pemFile, String message) {
...
+			return new char[0];
+		}
+
+		// get password bytes
+		var token = properties.getHash();
+		var decoder = Base64.getDecoder();
+		var index = Ed25519KeyStore.getIndex(pemFile);
+		var tokenBytes = decoder.decode(token);
+		if (tokenBytes.length < Constants.SALT_LENGTH + KEY_LENGTH / 8) {
+			logger.error("Token size check failed");
```
Comment (ljianghedera) [inline tools-ui/src/main/java/com/hedera/hashgraph/client/ui/utilities/KeyPairUtility.java:127]: If this happens, should the process continue or quit here? Seems to me it should quit immediately.

File: tools-ui/src/main/java/com/hedera/hashgraph/client/ui/utilities/KeyPairUtility.java [hunk 2]
```diff
@@ -73,18 +89,61 @@ private KeyPair getKeyPair(File pemFile, String message) {
...
+	 * 		the file where the pem file is stored;
+	 * @return the new password
+	 */
+	public static char[] resetPassword(String pemFile) throws HederaClientException, KeyStoreException {
+		var properties = new UserAccessibleProperties(DEFAULT_STORAGE + File.separator + USER_PROPERTIES, "");
+
+		// Check Hashcode
+		var storedHashCode = properties.getMnemonicHashCode();
+		var hashCode = Ed25519KeyStore.getMnemonicHashCode(pemFile);
+		if (hashCode != null && hashCode != storedHashCode) {
```
Comment (ljianghedera) [inline tools-ui/src/main/java/com/hedera/hashgraph/client/ui/utilities/KeyPairUtility.java:113]: nit: should the case `hashCode == null` be handled?

File: tools-ui/src/main/java/com/hedera/hashgraph/client/ui/utilities/KeyPairUtility.java (hunk 1 above)
Comment (DSMatusevich) [reply tools-ui/src/main/java/com/hedera/hashgraph/client/ui/utilities/KeyPairUtility.java:127]: good catch. will change

**Review by ljianghedera — APPROVED** [review]
LGTM

**Pull request merged**